pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -211,6 +212,39 @@ func GetMachineTypesFromCloudProfile(cloudProvider gardenv1beta1.CloudProvider,
 	return machineTypes
 }
 
+// GetMachineImagesFromCloudProfile retrieves a list of machine images from the cloud profile
+func GetMachineImagesFromCloudProfile(cloudProvider gardenv1beta1.CloudProvider, profile *gardenv1beta1.CloudProfile) []gardenv1beta1.MachineImage {
+	switch cloudProvider {
```

Comment:
Does it make sense to simplify the signature of this function to not provide the cloud provider? You could find it out automatically using https://github.com/gardener/gardener/blob/master/pkg/apis/garden/v1beta1/helper/helpers.go#L40.